src/ringct/bulletproofs.cc

```diff
@@ -647,17 +635,12 @@ bool bulletproof_VERIFY(const Bulletproof &proof)
   // PAPER LINES 21-22
   // The inner product challenges are computed per round
   rct::keyV w(rounds);
-  hashed.clear();
-  hashed.push_back(proof.L[0]);
-  hashed.push_back(proof.R[0]);
-  w[0] = rct::hash_to_scalar(hashed);
+  hash_cache_mash(hash_cache, proof.L[0]);
+  w[0] = hash_cache_mash(hash_cache, proof.R[0]);
   for (size_t i = 1; i < rounds; ++i)
   {
-    hashed.clear();
-    hashed.push_back(w[i-1]);
-    hashed.push_back(proof.L[i]);
-    hashed.push_back(proof.R[i]);
-    w[i] = rct::hash_to_scalar(hashed);
+    hash_cache_mash(hash_cache, proof.L[i]);
+    w[i] = hash_cache_mash(hash_cache, proof.R[i]);
```

Comment:
Can't the case for `w[0]` be also squashed into this for loop, as done on the prove side?